lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2265,8 +2273,14 @@ def has_different_resource_limits(self):
             memory=host_config.get('Memory'),
             memory_reservation=host_config.get('MemoryReservation'),
             memory_swap=host_config.get('MemorySwap'),
+            restart_policy=host_config.get('RestartPolicy')
```

Comment:
This is wrong. It should be `restart_policy=(host_config.get('RestartPolicy') or dict()).get('Name')`. But it's better to store `host_config.get('RestartPolicy') or dict()` into a new variable before, since you need to reuse it below.

Reply:
Nice! Very smart solution to guarantee compatibility. Adressed with 808987abac2e1e95f21db51def333bbb18ffd68e.